Jahia/search-ui-jahia-connector
# Backlog 14140 integrate tree facet in ui

https://jira.jahia.org/browse/BACKLOG-14140

## Description
Added integration for tree facets


## Tests

The following are included in this PR
- [x] Unit Tests (Most changes _should_ have unit tests)
   - ensure tree facet request is constructed properly
   - ensure that category facet data make sense for ui tree component
